diff --git a/wrtc/src/interfaces/native_network_interface.cpp b/wrtc/src/interfaces/native_network_interface.cpp
--- a/wrtc/src/interfaces/native_network_interface.cpp
+++ b/wrtc/src/interfaces/native_network_interface.cpp
@@ -18,7 +18,7 @@ namespace wrtc {
     void NativeNetworkInterface::initConnection(bool supportsPacketSending) {
         std::weak_ptr weak(shared_from_this());
-        networkThread().PostTask([weak, supportsPacketSending] {
+        networkThread().BlockingCall([weak, supportsPacketSending] {
             const auto strong = weak.lock();
             if (!strong) {
                 return;
             }
